### ISSUE TITLE
Natural::from_string_base fail to detect invalid character for long string

### ISSUE THREAD

**Issue by qingshi163** [mhogrefe/malachite#20]
the following code will fail
```
    assert_eq!(Natural::from_string_base(16, "1000000000000000z"), None);
```
but if reduce the length than will success.
```
    assert_eq!(Natural::from_string_base(16, "10000000z"), None);
```

**Comment by mhogrefe** [comment]
Thanks for the catch. The fix will be in the next release.

**Comment by mhogrefe** [comment]
Fixed in 0.4.0.
